pkg/querier/queryrange/query_range.go

```diff
@@ -222,6 +222,16 @@ func (prometheusCodec) DecodeRequest(_ context.Context, r *http.Request) (Reques
 	result.Query = r.FormValue("query")
 	result.Path = r.URL.Path
 
+	// Include the headers from http request in prometheusRequest.
+	// We don't include the following header in the codec -
+	// accept-encoding - defaultroundtripper uses the response_compression_enabled setting to inflate/deflate responses
```

Comment:
Please clarify which `defaultroundtripper` you are talking about - this one? https://github.com/prometheus/client_golang/blob/35ee299e50b8b3a40af7bdc9d9703a22dff0ecb6/api/client.go#L29

Is this behaviour documented?

Reply:
From our testing, we found that in the case of range queries only the client's encoding header if passed to the queries(non-cortex queriers when using Cortex in frontend-only mode) the responses expected by the defaultRoundtripper causes unexpected format and throws. It always expects a gzip response and returns it unzipped.

The behavior works fine if we ignore the accept-encoding header, kind of what it was doing before any headers were transmitted.  For the pipe to honor the accept-encoding header probably can be a separate issue and we can see if we can find the right fix for that. So at this moment, I feel we can keep the status-quo for the encoding header.

Reply:
When I said "please clarify" I meant in the code, so the comment can be understood without tracking down the PR where it was discussed.

Reply:
Ah I see. Done.